### PR TITLE
replace individual endpoint_cnt read from store with 1 bulk read

### PR DESCRIPTION
**Pull request by mavenugo**
getNetworksFromStore reads networks and endpoint_cnt from the kvstores.
endpoint_cnt especially is read in a for-loop for each network and that
causes a lot of stress in poorly performing KV-Stores.
This fix eases the load on the kvstore by fetching all the endpoint_cnt
in a single read and the operation is performed on it.

Signed-off-by: Madhu Venugopal <madhu@docker.com>

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -152,21 +153,24 @@ func (c *controller) getNetworksFromStore() ([]*network, error) {
 			continue
 		}
 
+		kvep, err := store.Map(datastore.Key(epCntKeyPrefix), &endpointCnt{})
```
Comment (aboch) [inline store.go]: Looking at the diffs, I am realizing there was no need for `getNetworks()` to reconstruct the epCount for each pulled network from store. Whatever code that makes a decision about the epCount, it does pull the latest epCount from store itself. There is no need to retrieve and construct the internal epCounts in the networks retrieved as list.

My advice is to simply remove the existing code which pulls the epCount in the loop below.
It is not needed.

Comment (aboch) [inline store.go]: Goal now is to improve the performance of current code in respect to the getNetworksFromStore().
Given` getNetworksFromStore()` retrieves the list of networks in a batch, it makes a lot of sense to retrieve the epCnt as a batch along it.

We can address the code removal I was suggesting in a subsequent PR outside of the release cycle.

File: endpoint_info.go [hunk 2]
```diff
@@ -181,21 +181,23 @@ type tableEntry struct {
...
-	if err != nil {
-		return nil
-	}
-
 	sb, ok := ep.getSandbox()
 	if !ok {
-		// endpoint hasn't joined any sandbox.
-		// Just return the endpoint
-		return ep
+		n, err := ep.getNetworkFromStore()
```
Comment (aboch) [inline endpoint_info.go]: Can you please push this optimization as a separate commit inside this PR.

Comment (mavenugo) [inline endpoint_info.go]: ok. will do

**Comment by aboch** [comment]
LGTM

**Pull request merged**